### PR TITLE
Fix panic when specifying a nil session ID

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Propagate the session ID selected by the broker when a nil session ID is
specified.  This requires an updated go-amqp to work.
Removed validation of EnqueuedSequenceNumber as it's not guaranteed to
be returned.

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Fix adapted from https://github.com/Azure/azure-sdk-for-js/blob/281b17ed42e29012be7089ae6f57a75e08633f62/sdk/servicebus/service-bus/src/session/messageSession.ts#L264

File: receiver.go [hunk 1]
```diff
@@ -408,33 +412,43 @@ func (r *Receiver) newSessionAndLink(ctx context.Context) error {
...
-	const name = "com.microsoft:session-filter"
 	const code = uint64(0x00000137000000C)
 
 	if !r.useSessions {
 		return nil, false
 	}
 
 	if r.sessionID == nil {
-		return amqp.LinkSourceFilter(name, code, nil), true
+		return amqp.LinkSourceFilter(sessionFilterName, code, nil), true
```
Comment (richardpark-msft) [inline receiver.go:448]: Looks like you don't need the 'if' here at all (ie, r.SessionID can just be passed unconditionally)

**Review by richardpark-msft — APPROVED** [review]
Looks good to me. One small spot where you could potentially eliminate an 'if'.

I also tend to be pretty generous with comments in the code, not sure how you feel about it. This path feels sufficiently esoteric that it could be useful.

File: receiver.go (hunk 1 above)
Comment (jhendrixMSFT) [reply receiver.go:448]: This is one of the cases of "when nil isn't nil" in Go, and comes into play with `interface{}`.  When passing `nil`, the type of the `value interface{}` param will be `nil`.  When passing `r.sessionID`, the type will be `*string`.  So while they both could in fact be `nil`, their underlying types are different.  This is important to the [marshal](https://github.com/Azure/go-amqp/blob/master/encode.go#L43) function in `go-amqp` which differentiates between underlying types.

**Pull request merged**